odoo/addons/base/tests/test_res_partner.py

```diff
@@ -332,6 +332,10 @@ def test_display_name_translation(self):
 
         self.assertEqual(child_contact.with_context(lang='fr_FR').display_name, 'Parent, Autre adresse')
 
+    def test_all_partner_category_name(self):
+        self.env['res.partner.category'].create({'name': 'lovely'})
+        self.assertEqual(len(self.env['res.partner.category'].name_search(limit=1)), 1)
```

Comment:
Since `_search_display_name` is overridden, it makes sense to test the `name_search`

Some questions, maybe nitpickings
Why test `name_search(limit=1)`? Is it the commonly used rpc in real world?
Can we `name_search('test_lov')`?
This test requires that database had no 'res.partner.category' before. I prefer if it can work with/without demo data.